llvm/lib/SYCLLowerIR/ModuleSplitter.cpp

```diff
@@ -816,6 +816,15 @@ void ModuleDesc::saveSplitInformationAsMetadata() {
         SpecConstantsPass::SPEC_CONST_DEFAULT_VAL_MODULE_MD_STRING);
 }
 
+ModuleDesc ModuleDesc::makeDummy() const {
+  ModuleDesc MD(CloneModule(getModule()));
```

Comment:
It looks like we are cloning the module two times: first time here and then during call to `ModuleDesc::makeDummyImageIR` within `saveModuleIR` and I don't think that we really need this first clone.

I suggest that we call `saveModuleIR` here immediately, thus saving a module copy (they are generally expensive).